Готов второй проект

nastyaprana opened the pull request:
Отправляю на ревью

nastyaprana committed:
Выполнила второй проект

nastyaprana committed:
Внесла правки по тестам

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+
+  <body>
+    <header class="header">
+      <nav class="header__menu">
```
**Отлично:**

Отлично, что использовали тег `nav` для списка ссылок навигации

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+  <body>
+    <header class="header">
+      <nav class="header__menu">
+        <ul class="header__links-list">
```
**Отлично:**

Отличное использование тега списка `ul > li`

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+      <nav class="header__menu">
+        <ul class="header__links-list">
+          <li class="header__links-list-item_no-bullit">
+            <a class="header__link_active" href="#">главная</a>
```
**Надо исправить:**

Нужна общая сущность на случай если активная ссылка сменится - `header__link header__link_active`

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+          alt="Логотип"
+        />
+      </a>
+      <address class="address">
```
**Отлично:**

Отлично, что использовали тег `address`

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+        <br /><span> 265, Амстердам</span>
+      </address>
+    </header>
+    <main>
```
**Отлично:**

Отлично, что для основного содержимого страницы использовали тег `main`

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+    </header>
+    <main>
+      <section class="cover">
+        <h1 class="cover__title">
```
**Отлично:**

Верно подобрали уровень заголовка, отлично что `h1` только один в проекте

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+          </span>
+        </h1>
+        <div class="cover__description">
+          <h3 class="cover__description-text">
```
**Надо исправить:**

Следует соблюдать уровни заголовков, согласно структуре страницы тут следует использовать заголовок уровня `h2`, [подробнее](https://doka.guide/html/h1-h6/)

**Можно лучше:** 

Для подзаголовков лучше применять тег `<p>`, т.к. через теги `<h1>—<h6>` сайт настраивается для выдачи в поисковиках по ключевым запросам пользователя, их лучше применять когда необходимо разделить большой объем текста, например, как это делают в книгах

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+            <a class="card-link">
+              <article class="card card_type_film">
+                <div class="card__title">
+                  <p>Фильм режиссёра, который бросил киношколу</p>
```
**Надо исправить:**

Тут должен быть заголовок. Уровень заголовка (h1-h6) нужно определить исходя из структуры страницы - `h3`

Проконтролируйте это тут и далее в списке для заголовков

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+          <li class="lots__card-list-item">
+            <a class="card-link">
+              <article class="card card_type_film">
+                <div class="card__title">
```
**Можно лучше:**

Этот элемент `div` - лишняя обертка. Ее можно удалить без вреда для верстки. Старайтесь избегать лишних элементов в разметке. Класс `card__title` можно установить самому заголовку

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+                <div class="card__title">
+                  <p>Фильм режиссёра, который бросил киношколу</p>
+                </div>
+                <div class="card__text">
```
**Можно лучше:**

Этот элемент `div` - лишняя обертка. Ее можно удалить без вреда для верстки. Старайтесь избегать лишних элементов в разметке. Класс `card__text` можно установить самому параграфу

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+      <nav class="footer__menu">
+        <ul class="footer__menu-list">
+          <li class="footer__menu-list-item">
+            <a class="footer__menu-link_active" href="#">главная</a>
```
**Надо исправить:**

Нужна общая сущность на случай если активная ссылка сменится - `footer__menu-link footer__menu-link_active`

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+              />
+            </a>
+          </li>
+          <li class="footer__links-youtube">
```
**Надо исправить:**

Пункты списка полностью одинаковые, используйте для них один класс - `footer__social-list-item`

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+              <img class="footer__social-icon" src="./images/vk.svg" alt="Вк" />
+            </a>
+          </li>
+          <li class="footer__links-youtube">
```
**Надо исправить:**

Пункты списка полностью одинаковые, используйте для них один класс - `footer__social-list-item`

Mililiya commented on index.html:
```diff
@@ -2,9 +2,221 @@
+          </li>
+        </ul>
+      </nav>
+      <div>
```
**Можно лучше:**

Этот элемент `div` - лишняя обертка. Ее можно удалить без вреда для верстки. Старайтесь избегать лишних элементов в разметке.

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+.cover {
+  background-image: linear-gradient(rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6)),
+    url(../images/cover.jpg);
+  height: 757px;
```
**Надо исправить:**

Блоку не нужна фиксированная высота, ее стоит заменить на минимальную 

В случае, если элемент, содержащий, например, текстовый контент ограничить по высоте - теряется масштабируемость. При редактировании (Добавлении/удалении символов) - нарушается визуальное отображение верстки, она начинает "ломаться". Старайтесь избегать фиксированных значений для height;

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+  list-style: circle;
+  font-size: 18px;
+  font-weight: 400;
+  font-family: Raleway;
```
**Надо исправить:**

Использовать ```font-family: 'Шрифт'```,  является не совсем корректным решением. 
 В том случае, если по какой-либо причине шрифты не загрузятся или будет использоваться браузер, не поддерживающий текущий формат шрифта - следует предусмотреть такой исход и добавить системные шрифты. 
 Например, sans-serif - `font-family: Raleway, sans-serif;`

Проконтролируйте это тут и далее в файле

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+}
+
+.no-letter-spacing {
+  margin-left: -25px;
```
**Надо исправить:**

Использования отрицательных отступов является не самой лучшей практикой. Особенно для достижения отображения по макету.
От подхода следует отказаться

Проконтролируйте это тут и далее в файле

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+  font-weight: 500;
+  color: #ffffff;
+  background-color: transparent;
+  width: 256px;
```
**Можно лучше:**

В данном случае будет лучше использовать фиксацию минимальной ширины для элемента, так в случае изменения длинны текста ширина блока сможет гибко растянуться.

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+  color: #ffffff;
+  background-color: transparent;
+  width: 256px;
+  height: 51px;
```
**Можно лучше:**

В данном случае будет лучше использовать фиксацию минимальной высоты для элемента, так в случае изменения размера шрифта высота блока сможет гибко растянуться.

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+  word-spacing: 18px;
+}
+
+.bet-button {
```
**Надо исправить:**

https://disk.yandex.ru/i/VrhojCSrhnog9Q у кнопки рендерится неверный шрифт. Необходимо добавить `font-family`. Обратите внимание, если по какой-либо причине шрифты не загрузятся или будет использоваться браузер, не поддерживающий текущий формат шрифта - следует предусмотреть такой исход и добавить системные шрифты. Например, sans-serif - `font-family: Raleway, sans-serif;`

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+  margin-right: 25px;
+}
+
+.card_type_film {
```
**Надо исправить:**

У карточек есть повторяющиеся правила, их следует установить общему блоку `card` чтобы избежать повторения у селекторов. У модификаторов должно остаться **только одно правило** - разное фоновое изображение

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+
+.card {
+  width: 334px;
+  height: 563px;
```
**Надо исправить:**

Не нужна фиксированная высота, ее стоит заменить на минимальную 

В случае, если элемент, содержащий, например, текстовый контент ограничить по высоте - теряется масштабируемость. При редактировании (Добавлении/удалении символов) - нарушается визуальное отображение верстки, она начинает "ломаться". Старайтесь избегать фиксированных значений для height;

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+  background-image: linear-gradient(rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6)),
+    url(../images/card-lot-01.jpg);
+  background-repeat: no-repeat;
+  height: 563px;
```
**Надо исправить:**

Не нужна фиксированная высота, ее стоит заменить на минимальную 

В случае, если элемент, содержащий, например, текстовый контент ограничить по высоте - теряется масштабируемость. При редактировании (Добавлении/удалении символов) - нарушается визуальное отображение верстки, она начинает "ломаться". Старайтесь избегать фиксированных значений для height;

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+  flex-grow: 1;
+}
+
+.lots__card-list-item:last-child .card__title {
```
**Надо исправить:**

https://disk.yandex.ru/i/t7meclUKyrh-Sg, https://disk.yandex.ru/i/To5iNDqs1JZbLw по макету, отступы во всех карточках полностью одинаковые, слева и справа по 40px, сверху 51px, снизу 48px, эти отступы можно установить заголовку и параграфу

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+  text-decoration: underline;
+  text-transform: uppercase;
+  font-weight: 700;
+  font-size: 22px !important;
```
**Надо исправить:**

Не следует использовать `!important` без необходимости

Использовать `!important` можно. И это действительно важный инструмент. Тем не менее, его стоит использовать с умом.

`!important` надо использовать проактивно, а не реактивно.

Это значит, что его можно использовать тогда, когда вы абсолютно уверены, что вам всегда будет нужно, чтобы это стиль имел приоритет, и вы знаете об этом заранее.

Например, в уверены в том, что вы всегда хотите видеть ошибки красными:
```css
.error-text{
    color:#c00!important;
}
```

Даже если сообщение об ошибке будет выведено внутри блока, в котором цвет текста синий, мы можем быть уверены, что ошибка останется красной. Мы всегда хотим сообщать об ошибке красным цветом, и поэтому мы сразу пишем `!important`.

А вот когда мы используем `!important` реактивно, то есть в ответ на возникшую проблему, когда мы запутались и вместо того, чтобы разобраться, прём напролом, тогда это плохо.

Реактивное использование `!important` не решает проблему, а только прячет её. Надо лечить болезнь, а не симптомы. Проблема никуда не делась, мы просто перекрыли её сверх-специфичным селектором, тогда как нужно было заняться рефакторингом и архитектурой.

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+  margin-bottom: 27px;
+}
+
+.lots__card-list {
```
**Надо исправить:**

Согласно заданию при добавлении в разметку еще одной карточки последняя должна переноситься на новую строку. Для этого этому классу нужно задать свойство flex-wrap в подходящем значении
https://doka.guide/css/flex-wrap/

При переносе доп карточки на новую строку, отступ по горизонтали не должен схлопываться, чтобы этого избежать, можно воспользоваться `gap: 25px;`

https://disk.yandex.ru/i/G2bpY0uif-9gMQ при удалении одной карточки, расстояние 25px между оставшимися должно сохраняться

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+  background-repeat: no-repeat;
+  height: 563px;
+  width: 334px;
+  background-position-x: -103px;
```
**Надо исправить:**

https://disk.yandex.ru/i/qEalt_xsccLIeQ при переполнении контентом изображение должно подстраиваться под высоту карточки, стоит воспользоваться значением `center`

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+  height: 563px;
+  width: 334px;
+  background-position-x: -103px;
+  background-size: 162%;
```
**Надо исправить:**

https://disk.yandex.ru/i/qEalt_xsccLIeQ при переполнении контентом изображение должно подстраиваться под высоту карточки, стоит воспользоваться значением `cover`

Mililiya commented on styles/style.css:
```diff
@@ -1 +1,372 @@
+
+.header__menu {
+  margin-right: 9px;
+}
```
**Можно лучше:**

Между селекторами следует сохранять одну пустую строку

**Проконтролируйте это тут и далее в файле** 

Можно воспользоваться автоматическим форматированием

На Windows Shift + Alt + F для автоматического форматирования
На Mac Shift + опции + F для автоматического форматирования
На Ubuntu Ctrl + Shift + Я для автоматического форматирования
можно пользоваться онлайн-сервисами для форматирования кода: https://webformatter.com/

Mililiya reviewed (commented):
Здравствуйте, Анастасия. 
Нужно развернуть общий комментарий, что бы увидеть его полностью ↓

Была проделана большая работа и она выполнена на хорошем уровне!

Что понравилось в работе:

- Отлично структурированный `html` код, страница валидна
- Страница соответствует макету
- Стили и изображения в собственных файлах и папках
- Используются технологии flex и grid.

Но есть так же и замечания, которые потребуют доработки.
Места, которые надо исправить в проекте:

- Комментарии в коде, отмеченные как "Надо исправить"

Что можно улучшить:

- Комментарии в коде, отмеченные как "Можно лучше"

Обратите внимание, комментарии "Надо исправить" обязательны для исправления, работа будет зачтена после исправления всех таких комментариев. Комментарии "Можно лучше" носят рекомендательный характер, их выполнение остается на Ваше усмотрение.

Вносите правки без спешки, следите за тем, что изменилось после внесения той или иной правки и при этом больше ничего не сломалось. Убедитесь перед отправкой, что все правки внесены, основные требования соответствуют брифу и чек листу. Если появятся трудности - обращайтесь к наставнику и старшему студенту, они Вам помогут.

Успешной работы над проектом, и до встречи на следующем ревью!

nastyaprana committed:
Внесла правки после ревью

1. Добавила общую сущность header__link header__link_active и footer__menu-link footer__menu-link_active в случае если активная ссылка сменится
2. Исправила уровни заголовков на корректные
3. Убрала лишние обертки div
4. Для одинаковых пунктов меню в футере использовала один класс footer__social-list-item
5. Заменила фиксированную высоту на минимальную
6. Добавила системные шрифты sans-serif - во всем файле
7. Убрала все отрицательные отступы
8. Использовала фиксацию минимальной ширины и высоты элементов в нужных блоках
9. Поправила шрифт у кнопки
10. Убрала повторяющиеся правила у карточек
11. Поправила отступы во всех карточках (они одинаковые)
12. Убрала !important там где в нем нет необходимости
13. Поправила карточки, при добавлении новой - она переносится на новую строку. Использовала flex-wrap и gap в данном случае.
14. Поправила подстройку изображений под высоту карточки (при переполнении контентом)
15. Сделала форматирование кода

Mililiya reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
